test_conformance/printf/util_printf.cpp

```diff
@@ -593,24 +741,27 @@ std::vector<printDataGenParameters> printVectorGenParameters = {
 
     //(Minimum)Two-wide,two positions after decimal
 
-    {NULL,"(1.0f,2.0f,3.0f,4.0f)","%2.2","hlf","float","4"},
+    { NULL, "(1.0f,2.0f,3.0f,4.0f)", "%2.2", "hlf", "float", "4" },
 
-    //Alternative form,uchar argument
+    // Alternative form,uchar argument
 
-    {NULL,"(0xFA,0xFB)","%#","hhx","uchar","2"},
+    { NULL, "(0xFA,0xFB)", "%#", "hhx", "uchar", "2" },
 
-    //Alternative form,ushort argument
+    // Alternative form,ushort argument
 
-    {NULL,"(0x1234,0x8765)","%#","hx","ushort","2"},
+    { NULL, "(0x1234,0x8765)", "%#", "hx", "ushort", "2" },
 
-  //Alternative form,uint argument
+    // Alternative form,uint argument
 
-    {NULL,"(0x12345678,0x87654321)","%#","hlx","uint","2"},
+    { NULL, "(0x12345678,0x87654321)", "%#", "hlx", "uint", "2" },
 
-    //Alternative form,long argument
+    // Alternative form,long argument
 
-    {NULL,"(12345678,98765432)","%","ld","long","2"}
+    { NULL, "(12345678,98765432)", "%", "ld", "long", "2" },
 
+    //(Minimum)Two-wide,two positions after decimal
+
+    { NULL, "(1.0h,2.0h,3.0h,4.0h)", "%2.2", "hlf", "half", "4" }
```

Comment:
Think rather than `hlf` (where `hl` seems to be for float) we should use `hf` instead:

> hl This modifier can only be used with the vector specifier. Specifies that a following d, i, o, u, x, or X conversion specifier applies to a intn or uintn argument; that a following a, A, e, E, f, F, g, or G conversion specifier applies to a floatn argument.

> h Specifies that a following d, i, o, u, x, or X conversion specifier applies to a shortn or ushortn argument (the argument will not be promoted); that a following a, A, e, E, f, F, g, or G conversion specifier applies to a halfn [62] argument.

cc- @frasercrmck  who found this

Reply:
I think all vector literals should be explicitly cast to the vector type. For example here we should have `(half4)((1.0h,2.0h,3.0h,4.0h))`.

> A vector literal is written as a parenthesized vector type followed by a parenthesized comma delimited list of parameters.

Reply:
> I think all vector literals should be explicitly cast to the vector type. For example here we should have `(half4)((1.0h,2.0h,3.0h,4.0h))`.
> 

I think the cast is happening already, based on the [kernel code](https://github.com/KhronosGroup/OpenCL-CTS/blob/0bcb99088def37a710da0459d81bab6921d4464b/test_conformance/printf/test_printf.cpp#L255-L276) which will create a variable called `tmp` with the cast type (e.g `half4`), and then `tmp` is passed to printf

Reply:
Right you are! Sorry for the noise.

Reply:
@EwanC I've applied `hf` correction. Test passes with Intel HDG driver.